rust/src/transformations/sum/int/monotonic/mod.rs

```diff
@@ -110,29 +102,13 @@ where
     )
 }
 
-#[doc(hidden)]
-/// Checks if two elements of type T have the same sign
-pub trait IsMonotonic: Sized {
-    fn is_monotonic(bounds: (Self, Self)) -> bool;
-}
-
-macro_rules! impl_same_sign_signed_int {
-    ($($ty:ty)+) => ($(impl IsMonotonic for $ty {
-        fn is_monotonic((a, b): (Self, Self)) -> bool {
-            a == 0 || b == 0 || (a > 0) == (b > 0)
-        }
-    })+)
-}
-impl_same_sign_signed_int! { i8 i16 i32 i64 i128 isize }
-
-macro_rules! impl_same_sign_unsigned_int {
-    ($($ty:ty)+) => ($(impl IsMonotonic for $ty {
-        fn is_monotonic(_: (Self, Self)) -> bool {
-            true
-        }
-    })+)
+/// # Proof Definition
+/// Returns true if and only if the signs of `a` and `b` agree,
+/// assuming that values of zero always agree.
+pub(crate) fn signs_agree<T: Number>((a, b): (T, T)) -> bool {
+    let _0 = T::zero();
+    a == _0 || b == _0 || (a > _0) == (b > _0)
```

Comment:
```suggestion
    (a == _0 && b == _0) || (a > _0) == (b > _0)
```
or
```suggestion
    (a == b) || (a > _0) == (b > _0)
```
Maybe I'm missing something, but it looks like this could return true if `a == 0`, and not even look at `b`, which doesn't seem right.

Reply:
This is the correct behavior for this function! If either value is zero, then the sign will always agree. That is, -4 and 0 are considered to have agreement in sign, as 0 agrees to anything.

Reply:
ah, got it:
> assuming that values of zero always agree